### PR TITLE
feature-set: core bpf stake program v5

### PR DESCRIPTION
**Pull request by 2501babe**
#### Problem
the [bpf stake v5](https://github.com/solana-foundation/solana-improvement-documents/pull/490) simd has been approved, so its agave time

#### Summary of Changes
impl it. the core bpf upgrade path seems extremely straightforward but @buffalojoec should double check i didnt miss anything. i declined to add this to `SVMFeatureSet` on purpose, i didnt see any plausible use for it

the places where i swap the new feature in for `stake_raise_minimum_delegation_to_1_sol` are all consensus-safe because both variables are unconditionally false (unless your name is "testnet"... but this will be corrected in the next restart)

Feature Gate Issue: https://github.com/anza-xyz/feature-gate-tracker/issues/326

### CODE REVIEW

File: program-runtime/src/invoke_context.rs [hunk 1]
```diff
@@ -672,12 +672,6 @@ impl<'a, 'ix_data> InvokeContext<'a, 'ix_data> {
             .program_runtime_environment_for_deployment
     }
 
-    pub fn is_stake_raise_minimum_delegation_to_1_sol_active(&self) -> bool {
-        self.environment_config
-            .feature_set
-            .stake_raise_minimum_delegation_to_1_sol
-    }
```
Comment (2501babe) [inline program-runtime/src/invoke_context.rs:681]: nothing in agave uses this so its better to remove lest anyone confuse it with a feature that actually matters

**Comment by nagisa** [comment]
Is there anywhere I could see the changes to the program itself?

**Comment by 2501babe** [comment]
@nagisa the [changelog](https://github.com/solana-program/stake/releases/tag/program%40v5.0.0) points to all pull requests new to this release

File: feature-set/src/lib.rs [hunk 2]
```diff
@@ -1311,6 +1311,14 @@ pub mod direct_account_pointers_in_program_input {
     solana_pubkey::declare_id!("ptrXWLkSDMZZmZN8GAT6W5yW4EvYByfw6cRRHbXwQNS");
 }
 
+pub mod core_bpf_stake_program_v5 {
```
Comment (buffalojoec) [inline feature-set/src/lib.rs]: nit: I would try to match the SIMD name a bit more, ie `upgrade_bpf_stake_program_to_v5`.

File: rpc/src/rpc.rs
Comment (buffalojoec) [inline rpc/src/rpc.rs:1]: Looks like there's another usage of the original minimum delegation feature in here still:
https://github.com/2501babe/solana/blob/5eea3e517b61141c57c84439c119bda5b1bc7c01/rpc/src/rpc.rs#L9237

**Review by buffalojoec — COMMENTED** [review]
I suppose leaving the original minimum delegation feature ID in the archival feature set is NBD (perhaps preferred?), but there shouldn't be any reason to keep it in `SVMFeatureSet`, right? If it's unused, let's remove it.
https://github.com/anza-xyz/agave/blob/90bf3bf14076f972fb571919f8c15aedee731232/svm-feature-set/src/lib.rs#L30

Also, in programs/sbf, there's a test program for querying the minimum delegation on the Stake program. The current on-chain Stake program, as well as the one in program-binaries will return 1 lamport, so the test program will become out of sync once we update program-binaries with v5.
https://github.com/solana-program/stake/blob/71276cb04195b388dc00804e6d0d913cb9de715f/program/src/lib.rs#L34-L38

Is this test program even used? Seems like maybe dead code.

> i declined to add this to `SVMFeatureSet` on purpose, i didnt see any plausible use for it

Yep, that's correct. Nothing SVM and below needs to key on the change.

File: rpc/src/rpc.rs
Comment (2501babe) [reply rpc/src/rpc.rs:1]: [27e5847](https://github.com/anza-xyz/agave/pull/11316/commits/27e584735adabc7812ec0b40cb3f93a0dbaa0925)

File: feature-set/src/lib.rs (hunk 2 above)
Comment (2501babe) [reply feature-set/src/lib.rs]: [27e5847](https://github.com/anza-xyz/agave/pull/11316/commits/27e584735adabc7812ec0b40cb3f93a0dbaa0925)

**Review by buffalojoec — APPROVED** [review]
LGTM

> Also, in programs/sbf, there's a test program for querying the minimum delegation on the Stake program. The current on-chain Stake program, as well as the one in program-binaries will return 1 lamport, so the test program will become out of sync once we update program-binaries with v5.
https://github.com/solana-program/stake/blob/71276cb04195b388dc00804e6d0d913cb9de715f/program/src/lib.rs#L34-L38
>
> Is this test program even used? Seems like maybe dead code.

This can be done in follow-up, but lmk!

**Comment by 2501babe** [comment]
>I suppose leaving the original minimum delegation feature ID in the archival feature set is NBD (perhaps preferred?)

my sense is that we never remove feature gates so that downstream can always confirm theyre active/inactive. if we did want to remove this one though we definitely cant until its for sure rolled back on testnet :sweat_smile: 

>but there shouldn't be any reason to keep it in SVMFeatureSet, right?

done

>Also, in programs/sbf, there's a test program for querying the minimum delegation on the Stake program. The current on-chain Stake program, as well as the one in program-binaries will return 1 lamport, so the test program will become out of sync once we update program-binaries with v5.

yep this is dead code. ill get it in a followup
